R/utilities-building-block.R

```diff
@@ -0,0 +1,177 @@
+#' values indicating whether the quantity is present or not. If a list is provided,
+#' the length of this list should be equal to the length of `quantityPaths`.
+#' @param negativeValuesAllowed A single boolean value or a list of boolean values
+#' indicating whether negative values are allowed for the quantities. If a list is provided,
+#' the length of this list should be equal to the length of `quantityPaths`.
+#'
+#' @param formula A list of `Formula` objects that will be set for the quantities.
+#' Mixing of `formula` and `quantityValues` is not allowed. If `formula` is provided,
+#' `quantityValues` should be `NULL`. The length of this list should be equal to the length of `quantityPaths`.
+#'
+#' @returns The updated `icBuildingBlock` object.
+#' TBD: no return? To be consistent (or not confusing) with the extend functions?
+#' @export
+#'
+#' @examples
+setInitialConditions <- function(icBuildingBlock,
+                                 quantityPaths,
+                                 quantityValues,
+                                 scaleDivisors = 1,
+                                 isPresent = TRUE,
+                                 negativeValuesAllowed = FALSE,
+                                 formula = NULL){
+
+  return(icBuildingBlock)
+}
+
+#' Delete entries from an Initial Conditions Building Block
+#'
+#' @param icBuildingBlock A `BuildingBlock` object of type `Initial Conditions`.
+#' @param quantityPaths A list of full paths of the quantities (usually molecules) that
+#' will be deleted. Entries not present in the provided BB are ignored. Should contain
+#' all path elements and the molecule name, separated by `|`.
+#'
+#' @export
+#'
+#' @examples
+deleteInitialConditions <- function(icBuildingBlock,
+                                    quantityPaths){
+}
+
+#' Extend an Initial Conditions Building Block (BB) with new entries for molecules
+#' from a molecules BB in all physical containers of a spatial structure BB.
+#'
+#' @param icBuildingBlock A `BuildingBlock` object of type `Initial Conditions`.
+#' @param spatialStructureBB A `BuildingBlock` object of type `Spatial Structure`.
+#' Entries will be created for the selected molecules in all physical containers of this
+#' spatial structure.
+#' @param moleculesBB A `BuildingBlock` object of type `Molecules`. The entries will be
+#' created for all molecules from this building block, or for a subset of molecules
+#' defined in the `moleculeNames` argument.
+#' @param moleculeNames Optional list of molecule names. If provided, only the molecules
+#' with these names will be added to the `icBuildingBlock`.
+#'
+#' @returns Paths of entries added to the building block.
+#' @export
+#'
+#' @examples
+extendInitialConditions <- function(icBuildingBlock,
+                                    spatialStructureBB,
+                                    moleculesBB,
+                                    moleculeNames = NULL){
```

Comment:
* should we also provide list of container paths (NULL per default) to extend only for selected containers?
* it is not clear from the comment what will happen if an initial condition for the combination {container, molecule} alreade exists in the IC-BB (overwritten, kept AS IS or error)

Reply:
> it is not clear from the comment what will happen if an initial condition for the combination {container, molecule} already exists in the IC-BB (overwritten, kept AS IS or error)

Same behavior as in MoBi -  I assume kept AS IS (== ignored), @rwmcintosh can you confirm?